core/app/models/spree/payment.rb

```diff
@@ -30,6 +30,7 @@ class Payment < Spree::Base
 
     # invalidate previously entered payments
     after_create :invalidate_old_payments
+    after_create :create_eligible_credit_event
```

Comment:
is there a way we can do this without callbacks? Overuse of callbacks is large problem in Spree, and should be curtailed